metadata-io/src/main/java/com/linkedin/metadata/service/DomainService.java

```diff
@@ -0,0 +1,354 @@
+package com.linkedin.metadata.service;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.linkedin.common.UrnArray;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.domain.Domains;
+import com.linkedin.entity.Aspect;
+import com.linkedin.metadata.Constants;
+import com.linkedin.metadata.resource.ResourceReference;
+import com.linkedin.mxe.MetadataChangeProposal;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import  com.linkedin.entity.client.EntityClient;
+import com.datahub.authentication.Authentication;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+
+import static com.linkedin.metadata.entity.AspectUtils.*;
+
+
+@Slf4j
+public class DomainService {
```

Comment:
does it make sense to create an interface for these tag/term/owner/domain services that they all implement?

Reply:
refactored common code into BaseService